examples/usb_serial.rs

```diff
@@ -0,0 +1,134 @@
+//! CDC-ACM serial port example using polling in a busy loop.
+//! This example currently requires an 8MHz external oscillator
+//! and assumed an LED is connected to port A6.
+//!
+//! Further work could be done to setup the HSI48 and the clock
+//! recovery system to generate the USB clock.
+
+#![no_std]
+#![no_main]
+
+use defmt_rtt as _;
+
+use hal::rcc::PllMDiv;
+use hal::rcc::PllNMul;
+use hal::rcc::PllQDiv;
+use hal::rcc::PllRDiv;
+use hal::usb::configure_usb_clock_source;
+use hal::usb::ClockSource;
+use panic_probe as _;
+
+use stm32g4 as _;
+
+#[cfg(feature = "defmt-logging")]
+#[defmt::panic_handler]
+fn panic() -> ! {
+    cortex_m::asm::udf()
+}
+
+pub fn exit() -> ! {
+    loop {
+        cortex_m::asm::bkpt();
+    }
+}
+
+use hal::rcc::{Config, PLLSrc, Prescaler};
+
+use hal::rcc::clock_recovery_system::{CrsConfig, CrsExt};
+
+use stm32g4xx_hal as hal;
+
+use hal::prelude::*;
+use hal::stm32;
+use hal::usb::{Peripheral, UsbBus};
+
+use usb_device::prelude::*;
+use usbd_serial::{SerialPort, USB_CLASS_CDC};
+
+#[cortex_m_rt::entry]
+fn main() -> ! {
+    let dp = stm32::Peripherals::take().unwrap();
+
+    let rcc = dp.RCC.constrain();
+
+    // This sets the clocks up as follows
+    // - 8 MHz external oscillator
+    // - Sysclck and HCLK at 144 MHz
+    // - APB1 = PCLK1 =  72 MHz
+    // - APB2 = PCLK2 =  72 MHz
+    // - USB = 48 MHz
+    let mut rcc = rcc.freeze(
+        Config::new(hal::rcc::SysClockSrc::HSE(8.mhz()))
+            .pll_cfg(hal::rcc::PllConfig {
+                mux: PLLSrc::HSE(8.mhz()),
+                m: PllMDiv::DIV_1,
+                n: PllNMul::MUL_36,
+                r: Some(PllRDiv::DIV_2),
+                q: Some(PllQDiv::DIV_6),
+                p: None,
+            })
+            .ahb_psc(Prescaler::Div2)
```

Comment:
Thought it was worth mentioning because I noticed in test, this div2 is setting the HCLK to 72Mhz -- differs from the comment above.